bindings/go/descriptor/descriptor.go

```diff
@@ -0,0 +1,177 @@
+	Resources          []*Resource             `json:"resources,omitempty"`
+	Sources            []*Source               `json:"sources,omitempty"`
+	References         []*Reference            `json:"componentReferences,omitempty"`
+}
+
+type ComponentIdentity struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+func (c *ComponentIdentity) String() string {
+	return fmt.Sprintf("%s:%s", c.Name, c.Version)
+}
+
+func ParseComponentIdentity(s string) (ComponentIdentity, error) {
+	ci := ComponentIdentity{}
+	split := strings.Split(s, ":")
+	if len(split) != 2 {
+		return ci, fmt.Errorf("invalid component identity %q", s)
+	}
+	ci.Name = split[0]
+	ci.Version = split[1]
+	if ci.Name == "" || ci.Version == "" {
+		return ci, fmt.Errorf("invalid component identity %q", s)
+	}
+	return ci, nil
+}
+
+func (c *ComponentIdentity) AsIdentity() runtime.Identity {
+	return runtime.Identity{
+		"name":    c.Name,
+		"version": c.Version,
+	}
+}
+
+type Resource struct {
+	ObjectMeta    `json:",inline"`
+	ExtraIdentity map[string]string `json:"extraIdentity,omitempty"`
+	SourceRefs    []*SourceRef      `json:"sourceRefs,omitempty"`
+	Type          string            `json:"type"`
+	Relation      string            `json:"relation"`
+	Access        Access            `json:"access"`
+	Digest        *Digest           `json:"digest,omitempty"`
+	Size          int64             `json:"size,omitempty"`
+}
+
+func (r *Resource) GetIdentity() map[string]string {
+	m := maps.Clone(r.ExtraIdentity)
+	if m == nil {
+		m = make(map[string]string)
+	}
+	m["name"] = r.Name
+	return m
+}
+
+type Source struct {
+	ObjectMeta    `json:",inline"`
+	ExtraIdentity map[string]string `json:"extraIdentity,omitempty"`
+	Type          string            `json:"type"`
+	Access        Access            `json:"access"`
+}
+
+func (r *Source) GetIdentity() map[string]string {
+	m := maps.Clone(r.ExtraIdentity)
+	if m == nil {
+		m = make(map[string]string)
+	}
+	m["name"] = r.Name
+	return m
+}
+
+type Access struct {
+	runtime.Typed `json:",inline"`
+}
+
+func (a *Access) UnmarshalJSON(data []byte) error {
+	a.Typed = &runtime.Raw{}
+	return json.Unmarshal(data, a.Typed)
+}
+
+func (a *Access) MarshalJSON() ([]byte, error) {
+	return json.Marshal(a.Typed)
+}
+
+type Reference struct {
+	Name          string            `json:"name"`
+	ExtraIdentity map[string]string `json:"extraIdentity,omitempty"`
+	Component     string            `json:"componentName"`
+	Version       string            `json:"version"`
+	Digest        Digest            `json:"digest,omitempty"`
+	Labels        []*Label          `json:"labels,omitempty"`
+}
+
+type SourceRef struct {
+	IdentitySelector map[string]string `json:"identitySelector,omitempty"`
+	Labels           map[string]string `json:"labels,omitempty"`
+}
+
+type Meta struct {
+	Version string `json:"schemaVersion"`
+}
+
+type ObjectMeta struct {
+	ComponentIdentity `json:",inline"`
```

Comment:
This surely should nto be a component identity for every object

Reply:
Indeed. Do we even need this? 🤔 Can't this just be `runtime.Identity`?

Reply:
I think the purpose was originally that it must be exactly name and version and can be converted to an identity. And then the rest of the fields would be extra identity